### PR TITLE
Add test and fix for null credentials causing NPE

### PR DESCRIPTION
**Pull request by NachoRaik**

### CODE REVIEW

File: src/main/java/com/mulesoft/extension/policies/authentication/AuthenticationOperations.java [hunk 1]
```diff
@@ -65,8 +66,9 @@ public void setAuthentication(@Optional String principal,
                                 @Optional String credentials,
                                 @Optional Map<String, Object> properties,
                                 AuthenticationHandler authenticationHandler) {
-    Authentication authentication =
-        new DefaultMuleAuthentication(new DefaultMuleCredentials(principal, credentials.toCharArray())).setProperties(properties);
+    Credentials authCredentials = new DefaultMuleCredentials(principal != null ? principal : "",
```
Comment (NachoRaik) [inline src/main/java/com/mulesoft/extension/policies/authentication/AuthenticationOperations.java]: Ain't there an annotation that allows me to handle this with a default value? `@NullSafe` does not work for Strings and the default value of `@Optional` can't be used
